lib/service/connect.go

```diff
@@ -822,11 +822,20 @@ func (process *TeleportProcess) findReverseTunnel(addrs []utils.NetAddr) (string
 
 		resp, err := clt.Find(process.ExitContext())
 		if err == nil {
-			// If a tunnel public address is set, return it otherwise return the
-			// tunnel listen address.
+			// Return the tunnel address in the following preference order:
+			//  1. Reverse Tunnel Public Address.
+			//  2. SSH Proxy Public Address.
+			//  3. HTTP Proxy Public Address.
+			//  4. Tunnel Listen Address.
 			if resp.Proxy.SSH.TunnelPublicAddr != "" {
 				return resp.Proxy.SSH.TunnelPublicAddr, nil
 			}
+			if resp.Proxy.SSH.SSHPublicAddr != "" {
+				return resp.Proxy.SSH.SSHPublicAddr, nil
```

Comment:
in this case, the port should be still updated to the listen port, not the SSH port, so utils.Host(SSHPublicAddr) + ":"+utils.Port(TunnelListenAddr) or default port